sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -261,7 +264,7 @@ public void AddTag(string name, object value)
                     _tagCollection?.Add(new KeyValuePair<string, object>(name, value!));
 #else
                     _tagCollection ??= new ActivityTagsCollection();
-                    _tagCollection.Add(name, value!);
+                    _tagCollection[name] = value!;
```

Comment:
Can we add a test that validates there are no dupes?

Reply:
I added one